controllers/discovery/initDiscoveryUserData.js

```diff
@@ -72,6 +89,7 @@ const InitDiscoveryUserData = async (req, res) => {
       type: QueryTypes.SELECT,
       replacements: {
         userId,
+        topicIds,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `topicIds` variable is being passed into the query replacements without any validation or sanitization. This could potentially lead to SQL injection if the `topicIds` are not properly validated. Ensure that `topicIds` are an array of integers before passing them to the query.

```diff
       type: QueryTypes.SELECT,
       replacements: {
         userId,
+        topicIds: Array.isArray(topicIds) ? topicIds.map(Number) : [],
         limit: limit,
         offset: page * limit
       }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->